src/merge.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  isCollection,
+  isDocument,
+  isMap,
+  isScalar,
+  isSeq,
+  isNode,
+  Node
+} from './nodes/Node'
+import { Document } from './doc/Document'
+import { Pair } from './nodes/Pair'
+import { visit } from './visit'
+import { Scalar } from './nodes/Scalar'
+import { Collection } from './nodes/Collection'
+import { YAMLMap } from './nodes/YAMLMap'
+import { YAMLSeq } from './nodes/YAMLSeq'
+import { MergeOptions } from './options'
+
+function getPath(ancestry: readonly (Document | Node | Pair)[]): (string|number)[] {
+  return ancestry.reduce<(string|number)[]>((p, { key }: any) => {
+    return (key !== undefined) ? [...p, key.value] : p;
+  }, []);
```

Comment:
This would be much clearer with a `for` loop filling out a previously initialised array. `reduce()` is rarely the right choice.

Reply:
I thought I had found the *one* use-case for `.reduce()`, but I agree. It's more readable using a for..of